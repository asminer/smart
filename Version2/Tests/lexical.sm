
// $Id$

/*
    A test of lexical ordering for the state space.
    Used only for display purposes.
*/

#PnMarkingStyle VECTOR

// Simple PN, look familiar?

pn molloy(bool mc, int n) := {
  // places, not in the usual order, to test lexical ordering
  place p1, p2, p3, p4, p5;
  init(p3:n);

  trans t1, t2, t3, t4, t5;
  cond(mc, 
    firing(t1 : expo(1.1), 
	   t2 : expo(2.1), 
	   t3 : expo(3.1), 
	   t4 : expo(4.1), 
	   t5 : expo(5.1)
	  ), 
    null
  );

  arcs(
	p3 : t1, t1 : p1, t1 : p4,
	p2 : t2, t2 : p1,
	p1 : t3, t3 : p2,
	p4 : t4, t4 : p5,
	p2 : t5, p5 : t5, t5 : p3
  );

  int ns := num_states(true);
  int na := num_arcs(true);
};

#StateDisplayOrder NATURAL

print("Natural order of states and reachability graph\n",
      sprint(molloy(false, 1).ns, " ", molloy(false, 1).na, "\n")
     );

print("Natural order of states and Markov chain\n",
      sprint(molloy(true, 1).ns, " ", molloy(true, 1).na, "\n")
     );



#StateDisplayOrder LEXICAL

print("Lexical order of states and reachability graph\n",
      sprint(molloy(false, 1).ns, " ", molloy(false, 1).na, "\n")
     );

print("Lexical order of states and Markov chain\n",
      sprint(molloy(true, 1).ns, " ", molloy(true, 1).na, "\n")
     );


