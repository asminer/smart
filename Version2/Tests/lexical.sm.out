Natural order of states and reachability graph
State 0: [0, 0, 1, 0, 0]
State 1: [1, 0, 0, 1, 0]
State 2: [0, 1, 0, 1, 0]
State 3: [1, 0, 0, 0, 1]
State 4: [0, 1, 0, 0, 1]
Reachability graph:
	To state 0
		From state 4
	To state 1
		From state 0
		From state 2
	To state 2
		From state 1
	To state 3
		From state 1
		From state 4
	To state 4
		From state 2
		From state 3
5 8
Natural order of states and Markov chain
State 0: [0, 0, 1, 0, 0]
State 1: [1, 0, 0, 1, 0]
State 2: [0, 1, 0, 1, 0]
State 3: [1, 0, 0, 0, 1]
State 4: [0, 1, 0, 0, 1]
Markov chain:
Column 0:
	4 : 5.1
Column 1:
	0 : 1.1
	2 : 2.1
Column 2:
	1 : 3.1
Column 3:
	1 : 4.1
	4 : 2.1
Column 4:
	2 : 4.1
	3 : 3.1
5 8
Lexical order of states and reachability graph
State 0: [0, 0, 1, 0, 0]
State 1: [0, 1, 0, 0, 1]
State 2: [0, 1, 0, 1, 0]
State 3: [1, 0, 0, 0, 1]
State 4: [1, 0, 0, 1, 0]
Reachability graph:
	To state 0
		From state 1
	To state 1
		From state 2
		From state 3
	To state 2
		From state 4
	To state 3
		From state 1
		From state 4
	To state 4
		From state 0
		From state 2
5 8
Lexical order of states and Markov chain
State 0: [0, 0, 1, 0, 0]
State 1: [0, 1, 0, 0, 1]
State 2: [0, 1, 0, 1, 0]
State 3: [1, 0, 0, 0, 1]
State 4: [1, 0, 0, 1, 0]
Markov chain:
Column 0:
	1 : 5.1
Column 1:
	2 : 4.1
	3 : 3.1
Column 2:
	4 : 3.1
Column 3:
	1 : 2.1
	4 : 4.1
Column 4:
	0 : 1.1
	2 : 2.1
5 8
